spark/src/test/scala/io/delta/tables/DeltaTableSuite.scala

```diff
@@ -234,9 +234,14 @@ class DeltaTableHadoopOptionsSuite extends QueryTest
   private val publicMethods =
   scala.reflect.runtime.universe.typeTag[io.delta.tables.DeltaTable].tpe.decls
     .filter(_.isPublic)
+    .filter(!_.isSynthetic)
```

Comment:
@Kimahriman Why do we need to add this filter here?

Reply:
The optional arguments on the clone methods create additional symbols that are like `clone$1`, `clone$2`, etc. This filters those extra ones out

Reply:
@Kimahriman What would be the problem if it is not filtered out?

Where do the additional symbols get created and placed?

Reply:
Removing it leads to:

```
DeltaTableHadoopOptionsSuite:
[info] io.delta.tables.DeltaTableHadoopOptionsSuite *** ABORTED ***
[info]   Set("cloneAtVersion$default$6", "cloneAtTimestamp$default$4", "cloneAtTimestamp$default$5", "clone$default$2", "clone$default$5", "cloneAtVersion$default$3", "cloneAtVersion$default$4", "clone$default$4", "cloneAtTimestamp$default$6", "cloneAtTimestamp$default$3", "cloneAtVersion$default$5", "clone$default$3") was not empty Found new methods added to DeltaTable: Set(cloneAtVersion$default$6, cloneAtTimestamp$default$4, cloneAtTimestamp$default$5, clone$default$2, clone$default$5, cloneAtVersion$default$3, cloneAtVersion$default$4, clone$default$4, cloneAtTimestamp$default$6, cloneAtTimestamp$default$3, cloneAtVersion$default$5, clone$default$3). Please make sure you add a new test to verify it works with file system options in this file, and update the `testedMethods` list. If this new method doesn't need to support file system options, you can add it to the `ignoredMethods` list (DeltaTableSuite.scala:266)
```

Reply:
@Kimahriman I see, I am working on adding the tests, once that's done we are able to remove the `filter`.